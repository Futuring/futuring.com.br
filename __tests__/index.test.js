/* global it, expect, describe */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import App from '../pages/index.js'

const photos = new Array(15).fill(0).map((v, k) => k + 1);
const url = { pathname: '/', query: {} };

describe('With Enzyme', () => {
  it('App have 15 photos', () => {
    const app = shallow(
      <App photos={photos} url={url}/>
    )

    expect(app.find('.photo').length).toBe(15);
  })
})

describe('With Snapshot Testing', () => {
  it('App have 15 photos', () => {
    const component = renderer.create(<App photos={photos} url={url}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})