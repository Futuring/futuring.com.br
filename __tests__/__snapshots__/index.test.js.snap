exports[`With Snapshot Testing App have 15 photos 1`] = `
<div
  className="list"
  data-jsx={3611748052}>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=1"
      onClick={[Function]}>
      1
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=2"
      onClick={[Function]}>
      2
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=3"
      onClick={[Function]}>
      3
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=4"
      onClick={[Function]}>
      4
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=5"
      onClick={[Function]}>
      5
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=6"
      onClick={[Function]}>
      6
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=7"
      onClick={[Function]}>
      7
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=8"
      onClick={[Function]}>
      8
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=9"
      onClick={[Function]}>
      9
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=10"
      onClick={[Function]}>
      10
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=11"
      onClick={[Function]}>
      11
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=12"
      onClick={[Function]}>
      12
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=13"
      onClick={[Function]}>
      13
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=14"
      onClick={[Function]}>
      14
    </a>
  </div>
  <div
    className="photo"
    data-jsx={3611748052}>
    <a
      className="photoLink"
      data-jsx={3611748052}
      href="/photo?id=15"
      onClick={[Function]}>
      15
    </a>
  </div>
</div>
`;
